lib/hooks/core/hook-http.js

```diff
@@ -44,7 +44,10 @@ function requestWrap(request) {
     var root = cls.getRootContext();
 
     if (!root) {
-      agent.logger.debug('Untraced http options:', options);
+      if (typeof options === 'string') {
+        options = url.parse(options);
```

Comment:
Very very minor nit: get rid of this assignment and conditional, and inline the whole expression into the call to `uriFromOptions`.

This would allow the optimizer to avoid having to merge type-information about `option` from a rarely (or never taken) path with the mainline path.